shared/common-adapters/input.native.js

```diff
@@ -0,0 +1,119 @@
+/* @flow */
+
+import React, {Component} from 'react'
+import {TextInput} from 'react-native'
+import {globalColors, globalStyles} from '../styles/style-guide'
+import Platform, {OS} from '../constants/platform'
+import Text from './text.native'
+import Box from './box'
+
+import type {Props} from './input'
+
+type State = {
+  inputFocused: boolean,
+  text: string
+}
+
+// TODO figure out how to make a passwordVisible type
+// We don't want to set this to clear text because that gives
+// permission to the OS to let the text be saved to use for predictive typing.
+
+class Input extends Component<void, Props, State> {
+  state: State;
+  _textInput: any;
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      inputFocused: false,
+      text: this.props.value || ''
+    }
+  }
+
+  _setPasswordVisible (props: Props) {
+    const passwordVisible = props.type === 'passwordVisible'
+    if (this._textInput) {
+      this._textInput.setNativeProps({passwordVisible})
+    }
+  }
+
+  componentWillUpdate (nextProps: Props) {
+    if (nextProps.type !== this.props.type) {
+      this._setPasswordVisible(nextProps)
+    }
+  }
+
+  render () {
+    const IOS = Platform.OS_IOS === OS
+    const inputStyle = Text.textStyle({type: 'BodySemibold'}, {})
+    const password = this.props.type === 'password'
+    const passwordVisible = this.props.type === 'passwordVisible'
+
+    return (
+      <Box style={{...containerStyle, ...this.props.style}}>
+        {this.state.text.length > 0 && <Text type='BodySmall' style={{...floatingLabelStyle}}>{this.props.floatingLabelText}</Text>}
+        <TextInput
+          style={{...inputStyle, ...textInputStyle}}
+          ref={component => { this._textInput = component }}
+          autoCorrect={!(password || passwordVisible)}
+          defaultValue={this.props.value}
+          secureTextEntry={password}
+          autoFocus={this.props.autoFocus}
+          placeholder={this.props.hintText}
+          placeholderColor={globalColors.black10}
+          underlineColorAndroid={this.state.inputFocused ? globalColors.blue : globalColors.black10}
+          onFocus={() => this.setState({inputFocused: true})}
+          onBlur={() => this.setState({inputFocused: false})}
+          onSubmitEditing={this.props.onEnterKeyDown}
+          onChange={this.props.onChange}
+          onChangeText={text => this.setState({text})} />
```

Comment:
that seems right, it might make sense to add that to the type for Input and also implement it on desktop